src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -165,6 +165,7 @@ export const Configurator = ({
             onConfirm(dataset);
           }}
           fill
+          disabled={timeFields && timeFields.length > 0 && timeFieldName === undefined}
```

Comment:
nit: but couldn't you combine the first two expressions down to `timeFields?.length > 0`. 
do you think that the way you did it would be better style?

Reply:
This will conflict with your time field fixes PR. use the logic from there.